src/overlays/actors/ovl_Obj_Kibako/z_obj_kibako.c

```diff
@@ -27,33 +36,242 @@ const ActorInit Obj_Kibako_InitVars = {
+}
+
+void ObjKibako_AirBreak(ObjKibako* this, GlobalContext* globalCtx) {
+    s16 angle;
+    s32 i;
+    Vec3f* pos = &this->actor.posRot.pos;
+    Vec3f spC8;
+    Vec3f spBC;
+
+    for (i = 0, angle = 0; i < 12; i++, angle += 0x4E20) {
+        f32 sn = Math_Sins(angle);
+        f32 cs = Math_Coss(angle);
+        f32 temp_rand;
+        s16 phi_s0;
+
+        spC8.x = sn * 16.0f;
+        spC8.y = (Math_Rand_ZeroOne() * 5.0f) + 2.0f;
+        spC8.z = cs * 16.0f;
+        spBC.x = spC8.x * 0.2f;
+        spBC.y = (Math_Rand_ZeroOne() * 6.0f) + 2.0f;
+        spBC.z = spC8.z * 0.2f;
+        spC8.x += pos->x;
+        spC8.y += pos->y;
+        spC8.z += pos->z;
+        temp_rand = Math_Rand_ZeroOne();
+        if (temp_rand < 0.1f) {
+            phi_s0 = 0x60;
+        } else if (temp_rand < 0.7f) {
+            phi_s0 = 0x40;
+        } else {
+            phi_s0 = 0x20;
+        }
+        // EffectSsKakera_Spawn
+        func_80029E8C(globalCtx, &spC8, &spBC, pos, -200, phi_s0, 10, 10, 0, (Math_Rand_ZeroOne() * 30.0f) + 10.0f, 0,
+                      32, 60, -1, 3, D_05005380);
+    }
+    func_80033480(globalCtx, &this->actor.posRot.pos, 40.0f, 3, 50, 140, 1);
+}
+
+void ObjKibako_WaterBreak(ObjKibako* this, GlobalContext* globalCtx) {
+    s16 angle;
+    s32 i;
+    Vec3f* pos = &this->actor.posRot.pos;
+    Vec3f spC8;
+    Vec3f spBC;
+
+    spC8 = *pos;
+    spC8.y += this->actor.waterY;
+    func_8002949C(globalCtx, &spC8, 0, 0, 0, 0x1F4);
+
+    for (i = 0, angle = 0; i < 12; i++, angle += 0x4E20) {
+        f32 sn = Math_Sins(angle);
+        f32 cs = Math_Coss(angle);
+        f32 temp_rand;
+        s16 phi_s0;
+
+        spC8.x = sn * 16.0f;
+        spC8.y = (Math_Rand_ZeroOne() * 5.0f) + 2.0f;
+        spC8.z = cs * 16.0f;
+        spBC.x = spC8.x * 0.18f;
+        spBC.y = (Math_Rand_ZeroOne() * 4.0f) + 2.0f;
+        spBC.z = spC8.z * 0.18f;
+        spC8.x += pos->x;
+        spC8.y += pos->y;
+        spC8.z += pos->z;
+        temp_rand = Math_Rand_ZeroOne();
+        phi_s0 = (temp_rand < 0.2f) ? 0x40 : 0x20;
+        // EffectSsKakera_Spawn
+        func_80029E8C(globalCtx, &spC8, &spBC, pos, -180, phi_s0, 30, 30, 0, (Math_Rand_ZeroOne() * 30.0f) + 10.0f, 0,
+                      32, 70, -1, 3, D_05005380);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Kibako/func_80B94D34.s")
+void ObjKibako_SetupIdle(ObjKibako* this) {
+    this->actionFunc = ObjKibako_Idle;
+    this->actor.colChkInfo.mass = 0xFE;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Kibako/ObjKibako_Init.s")
+void ObjKibako_Idle(ObjKibako* this, GlobalContext* globalCtx) {
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        ObjKibako_SetupHeld(this);
+    } else if ((this->actor.bgCheckFlags & 0x20) && (this->actor.waterY > 19.0f)) {
+        ObjKibako_WaterBreak(this, globalCtx);
+        Audio_PlaySoundAtPosition(globalCtx, &this->actor.posRot.pos, 0x14, NA_SE_EV_WOODBOX_BREAK);
```

Comment:
The duration argument in `Audio_PlaySoundAtPosition` should ideally be decimal. (relevant throughout the file)
```suggestion
        Audio_PlaySoundAtPosition(globalCtx, &this->actor.posRot.pos, 20, NA_SE_EV_WOODBOX_BREAK);
```